tools/clang/lib/SPIRV/SpirvEmitter.cpp

```diff
@@ -1118,10 +1118,16 @@ void SpirvEmitter::doFunctionDecl(const FunctionDecl *decl) {
     uint32_t flags = 3u;
     // The line number in the source program at which the function scope begins.
     auto scopeLine = sm.getPresumedLineNumber(decl->getBody()->getLocStart());
-    spvBuilder.createDebugFunction(funcName, source, line, column, parentScope,
-                                   funcName, flags, scopeLine, func);
+    SpirvDebugInstruction *debugFunction;
+    debugFunction = spvBuilder.createDebugFunction(
+        funcName, source, line, column, parentScope, funcName, flags, scopeLine,
+        func);
+
+    info->scopeStack.push_back(debugFunction);
   }
 
+  // TODO: If `decl->hasBody() == false`, add DebugFunctionDeclaration.
```

Comment:
I think the way the SPIR-V backend "discovers" functions is as follows:
it starts with the entry point function. Inside that function anytime it faces a function call, it'll find the function definition and add it to the queue of functions that must be visited. Therefore, I think we'll never visit a function declaration on its own. So, this TODO can be removed.